plugins/inputs/intel_powerstat/intel_powerstat.go

```diff
@@ -0,0 +1,493 @@
+			p.addCurrentDramPowerConsumption(socketID, acc)
+		}
+	}
+}
+
+func (p *PowerStat) addThermalDesignPowerMetric(socketID string, acc telegraf.Accumulator) {
+	maxPower, err := p.rapl.GetConstraintMaxPower(socketID)
+	if err != nil {
+		p.Log.Errorf("error while retrieving TDP of the socket %s, err: %v", socketID, err)
+		return
+	}
+
+	tags := map[string]string{
+		"package_id": socketID,
+	}
+
+	fields := map[string]interface{}{
+		"name":  "thermal_design_power",
+		"unit":  "Watt",
+		"value": services.RoundFloatToNearestTwoDecimalPlaces(services.ConvertMicroWattToWatt(maxPower)),
+	}
+
+	acc.AddGauge("powerstat_package", fields, tags)
+}
+
+func (p *PowerStat) addCurrentSocketPowerConsumption(socketID string, acc telegraf.Accumulator) {
+	tags := map[string]string{
+		"package_id": socketID,
+	}
+
+	fields := map[string]interface{}{
+		"name":  "current_power_consumption",
+		"unit":  "Watt",
+		"value": services.RoundFloatToNearestTwoDecimalPlaces(p.rapl.GetCurrentPackagePowerConsumption(socketID)),
+	}
+
+	acc.AddGauge("powerstat_package", fields, tags)
+}
+
+func (p *PowerStat) addCurrentDramPowerConsumption(socketID string, acc telegraf.Accumulator) {
+	tags := map[string]string{
+		"package_id": socketID,
+	}
+
+	fields := map[string]interface{}{
+		"name":  "current_dram_power_consumption",
+		"unit":  "Watt",
+		"value": services.RoundFloatToNearestTwoDecimalPlaces(p.rapl.GetCurrentDramPowerConsumption(socketID)),
+	}
+
+	acc.AddGauge("powerstat_package", fields, tags)
+}
+
+func (p *PowerStat) verifyProcessor() error {
+	stats, err := p.fs.GetCPUInfoStats()
+	p.cpuInfo = stats
+
+	if err != nil {
+		return err
+	}
+
+	// First CPU is sufficient for verification.
+	firstCPU := p.cpuInfo["0"]
+	if firstCPU == nil {
+		return fmt.Errorf("first core not found while parsing /proc/cpuinfo")
+	}
+
+	if firstCPU.VendorID != "GenuineIntel" || firstCPU.CPUFamily != "6" {
+		return fmt.Errorf("Intel processor not found, vendorId: %s", firstCPU.VendorID)
+	}
+
+	// Disable c1CoreStateResidency and c6CoreStateResidency for specific processors.
+	allowedProcessors, err := services.ConvertHexArrayToIntegerArray(allowedProcessorModelsForC1C6)
+	if err != nil {
+		return err
+	}
+	if !contains(services.ConvertIntegerArrayToStringArray(allowedProcessors), firstCPU.Model) {
+		p.c1CoreStateResidency = false
+		p.c6CoreStateResidency = false
+	}
+
+	if !strings.Contains(firstCPU.Flags, "msr") {
+		p.coreCPUTemperature = false
+		p.c6CoreStateResidency = false
+		p.processorBusyCycles = false
+		p.processorBusyFrequency = false
+		p.c1CoreStateResidency = false
+	}
+
+	if !strings.Contains(firstCPU.Flags, "aperfmperf") {
+		p.processorBusyFrequency = false
+		p.processorBusyCycles = false
+		p.c1CoreStateResidency = false
+	}
+
+	if !strings.Contains(firstCPU.Flags, "dts") {
+		p.coreCPUTemperature = false
+	}
+
+	return nil
+}
+
+// ParseCPUMetricsConfig enables specific metrics based on config values.
+func (p *PowerStat) ParseCPUMetricsConfig() {
```

Comment:
This should be a private function `parseCPUMetricsConfig()`

Reply:
Fixed.